aiida_zeopp/workflows/geometry.py

```diff
@@ -0,0 +1,112 @@
+CifData = DataFactory('cif')
+SinglefileData = DataFactory('singlefile')
+
+options = {
+    "resources": {
+        "num_machines": 1,
+        "tot_num_mpiprocs": 1,
+    },
+    "max_wallclock_seconds": 30 * 60,
+    "withmpi": False,
+}
+
+class ZeoppGeometryWorkChain(WorkChain):
+    """Workchain that for a computing geometry properties using zeo++."""
+
+    @classmethod
+    def define(cls, spec):
+        """Define inputs, logic and outputs of ZeoppGeometryWorkChain"""
+        super(ZeoppGeometryWorkChain, cls).define(spec)
+        
+        # Define the inputs, specifying the type we expect
+        spec.input("probe_radius", valid_type=Float, required=True)
+        spec.input("structure", valid_type=CifData, required=True)
+        spec.input("zeopp_code", valid_type=Code, required=True)
+        spec.input("_options", valid_type=dict, required=False, default=options)
+
+        # Define the outputs, specifying the type we expect
+        spec.output("block", valid_type=SinglefileData, required=False)
+        spec.output("pore_volume_volpo", valid_type=ParameterData)
+        spec.output("surface_area_sa", valid_type=ParameterData)
+        spec.output("free_sphere_res", valid_type=ParameterData)
+
+		# Define workflow logic
+        spec.outline(
+            cls.run_geom_zeopp,
+            if_(cls.should_run_block_zeopp)(
+                cls.run_block_zeopp,
+            ),
+            cls.return_result,
+        )
+
+    def run_geom_zeopp(self):
+        """This function will perform geometry analysis using zeo++."""
+        NetworkParameters = DataFactory('zeopp.parameters')
+        sigma = self.inputs.probe_radius.value
+        params = {
+            'ha': True,
+            'res': True,
+            'sa': [sigma, sigma, 10000],
+            'volpo': [sigma, sigma, 10000],
+        }
+        inputs = {
+            'code'       : self.inputs.zeopp_code,
+            'structure'  : self.inputs.structure,
+            'parameters' : NetworkParameters(dict=params),
+            '_options'   : self.inputs._options,
+            '_label'     : "run_geom_zeopp",
+        }
+
+        # Create the calculation process and launch it
+        future  = submit(ZeoppCalculation.process(), **inputs)
+        self.report("pk: {} | Running geometry analysis with zeo++".format(future.pid))
+        return ToContext(zeopp_geometry=Outputs(future))
+
+    def should_run_block_zeopp(self):
+        """If the pore non-accessible volume is 0 - there is no need to run block pocket calculation."""
+        return self.ctx.zeopp_geometry["pore_volume_volpo"].dict.PONAV_Volume_fraction  > 0.001
+        
+    def run_block_zeopp(self):
+        """This is the main function that will perform a zeo++ block pocket calculation."""
+        NetworkParameters = DataFactory('zeopp.parameters')
+        sigma = self.inputs.probe_radius.value
+        params = {
+            'ha':True,
+            'block': [sigma, 200],
```

Comment:
Is the value of 200 generic enough to work in (almost) all cases?
If yes, please add a comment here explaining how you chose it.

If not, perhaps one could
 * do some logic here to provide a guess adapted to the particular case (that would be most in the spirit of workflows).
 * provide another input dict `_settings` to the workflow where one can specify it.